jax/_src/basearray.pyi

```diff
@@ -158,7 +161,7 @@ class Array(abc.ABC):
   def flatten(self, order: str = "C") -> Array: ...
   @property
   def imag(self) -> Array: ...
-  def item(self, *args: int) -> Any: ...
+  def item(self, *args: int) -> bool | int | float | complex: ...
```

Comment:
I tried this in #23094 and ended up having to revert it because it led to so many downstream failures.

Reply:
Okay, thanks for letting me know. I reverted this.